connect/runtime/src/main/java/org/apache/kafka/connect/runtime/errors/RetryWithToleranceOperator.java

```diff
@@ -90,27 +72,31 @@
         TOLERABLE_EXCEPTIONS.put(Stage.VALUE_CONVERTER, Exception.class);
     }
 
+    private final long errorRetryTimeout;
+    private final long errorMaxDelayInMillis;
+    private final ToleranceType errorToleranceType;
+
     private long totalFailures = 0;
     private final Time time;
-    private RetryWithToleranceOperatorConfig config;
     private ErrorHandlingMetrics errorHandlingMetrics;
 
     protected ProcessingContext context = new ProcessingContext();
 
-    public RetryWithToleranceOperator() {
-        this(new SystemTime());
+    public RetryWithToleranceOperator(ConnectorConfig connConfig) {
+        this(connConfig, Time.SYSTEM);
     }
 
     // Visible for testing
-    public RetryWithToleranceOperator(Time time) {
-        this.time = time;
+    public RetryWithToleranceOperator(ConnectorConfig connConfig, Time time) {
+        this(connConfig.errorRetryTimeout(), connConfig.errorMaxDelayInMillis(), connConfig.errorToleranceType(), time);
```

Comment:
Could we keep just one of these constructors? We could keep the constructor that takes pass a config and have the tests pass a mock with mocked methods for these getters, or we keep the last form and change where this constructor is called. 

I think it would prefer the latter. We have a number of classes that are similar, so there is lots of prior art.